module/zfs/ddt_zap.c

```diff
@@ -109,22 +109,23 @@ ddt_zap_lookup(objset_t *os, uint64_t object, ddt_entry_t *dde)
 	uint64_t one, csize;
 	int error;
 
-	cbuf = kmem_alloc(sizeof (dde->dde_phys) + 1, KM_SLEEP);
-
 	error = zap_length_uint64(os, object, (uint64_t *)&dde->dde_key,
 	    DDT_KEY_WORDS, &one, &csize);
 	if (error)
-		goto out;
+		return (error);
 
 	ASSERT(one == 1);
 	ASSERT(csize <= (sizeof (dde->dde_phys) + 1));
 
+	cbuf = kmem_alloc(csize, KM_SLEEP);
+
 	error = zap_lookup_uint64(os, object, (uint64_t *)&dde->dde_key,
 	    DDT_KEY_WORDS, 1, csize, cbuf);
 	if (error)
 		goto out;
 
 	ddt_zap_decompress(cbuf, dde->dde_phys, csize, sizeof (dde->dde_phys));
```

Comment:
Ahh, it did change to `csize` in a later commit, which I've moved back down into the earlier "make allocation consistent" commit.

Re out: yes, very nice.

Reply:
Resolved in last push.

(I think I can't resolve comments because I'm not the original PR owner. If there's something we can press to make me "own" these, I can tidy up as I go, otherwise here we are!)

Reply:
I don't think there's any way for you to own them, I'll just resolve them as I work through your comments